entry/src/entry.rs

```diff
@@ -548,26 +547,22 @@ fn start_verify_transactions_gpu(
                     num_transactions,
                     "entry-sig-verify",
                 );
-                // We use set_len here instead of resize(num_txs, Packet::default()), to save
-                // memory bandwidth and avoid writing a large amount of data that will be overwritten
-                // soon afterwards. As well, Packet::default() actually leaves the packet data
-                // uninitialized, so the initialization would simply write junk into
-                // the vector anyway.
-                unsafe {
-                    packet_batch.set_len(num_transactions);
-                }
+
+                let uninitialized_packets = packet_batch.spare_capacity_mut().iter_mut();
```

Comment:
don't we need `assume_init` somewhere below?

Reply:
We do not, `set_len()` does the work for us at the end:
- `assume_init()` yields a `T` from a `MaybeUninit<T>`
- We are starting with a `Vec<Packet>` (the type under the hood for `PacketBatch`)
- `packet_batch.spare_capacity_mut()` allows us to access elements at index `i` where `vec_length <= i < vec_capacity`
    - But, our access to these is of type [`MaybeUninit<Packet>`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.spare_capacity_mut)
- We initialize the elements in place, so calling `set_len()` is saying "these are valid elements of the Vec now and can be accessed normally; also drop them normally when dropping the Vec`